ecl/hql/hqlutil.hpp

```diff
@@ -153,9 +153,19 @@ extern HQL_API bool isOpRedundantForCompare(IHqlExpression * expr);
 
 extern HQL_API bool isLengthPreservingCast(IHqlExpression * expr);
 
+/**
+ * Report fields that are present in one record but not in another
+ *
+ * @param newRecord     The record in which to search
+ * @param oldRecord     The record providing the fields to search for
+ * @param errs          Where to report errors
+ * @param location      Location to use when reporting errors
+ */
+extern HQL_API void reportDroppedFields(IHqlExpression * newRecord, IHqlExpression * oldRecord, IErrorReceiver &err, ECLlocation &location);
+
 extern HQL_API IHqlExpression * createTransformFromRow(IHqlExpression * expr);
 extern HQL_API IHqlExpression * createNullTransform(IHqlExpression * record);
-extern HQL_API IHqlExpression * createMappingTransform(IHqlExpression * selfSelector, IHqlExpression * inSelector);
+extern HQL_API IHqlExpression * createMappingTransform(IHqlExpression * selfSelector, IHqlExpression * inSelector, bool replaceMissingWithDefault, IErrorReceiver &err, ECLlocation &location);
```

Comment:
It creates a transform that maps from a row inSelector to a target selfSelector.  If replaceMissingWithDefault is true the source can be a subset of the target, and the default values for the fields are used to initialize the target fields.